lib/elixir/lib/kernel/typespec.ex

```diff
@@ -75,12 +75,21 @@ defmodule Kernel.Typespec do
 
   def spec_to_callback(module, {name, arity} = signature)
       when is_atom(module) and is_atom(name) and arity in 0..255 do
-    {_set, bag} = :elixir_module.data_tables(module)
+    {set, bag} = :elixir_module.data_tables(module)
 
     filter = fn {:spec, expr, pos} ->
       if spec_to_signature(expr) == signature do
-        delete_typespec(bag, :spec, expr, pos)
-        store_typespec(bag, :callback, expr, pos)
+        kind = :callback
+        store_typespec(bag, kind, expr, pos)
+
+        case :ets.lookup(set, {:function, name, arity}) do
```

Comment:
I tried locally just by removing the line `delete_typespec(bag, :spec, expr, pos)` and the `@spec` is shown on IEx, why is this part needed? 🙂